### ISSUE TITLE
proper client integration tests with the various jackson-versions

### ISSUE THREAD

**Issue by rymurr** [projectnessie/nessie#636]
As per #630 we have added tests for the client against various versions of Jackson. We have not yet tested all model objects and should run integration tests against the full REST api for different jackson versions

**Comment by rymurr** [comment]
This should also ensure we check against java 8 to take care of #637

**Comment by snazy** [comment]
@rymurr do we still need this one?

**Comment by rymurr** [comment]
yeah, we only check the fact that the http client works, not that the full nessie api w/ jackson annotations works

**Comment by harshm-dremio** [comment]
@rymurr 
In context of [636](https://github.com/projectnessie/nessie/pull/636), 

>  We have not yet tested all model objects and should run integration tests against the full REST api for different jackson versions

Few questions - 

Q1 - Are you referring to these or more ?
```
./org/projectnessie/client/TestResultStreamPaginator.java
./org/projectnessie/client/util/JaegerTestTracer.java
./org/projectnessie/client/util/TestServer.java
./org/projectnessie/client/auth/TestNoneAuthProvider.java
./org/projectnessie/client/auth/TestAwsAuthProvider.java
./org/projectnessie/client/auth/TestBasicAuthProvider.java
./org/projectnessie/client/auth/TestBearerAuthenticationProvider.java
./org/projectnessie/client/TestNessieHttpClient.java
./org/projectnessie/client/http/TestHttpClientBuilder.java
./org/projectnessie/client/http/TestUriBuilder.java
./org/projectnessie/client/http/TestHttpsClient.java
./org/projectnessie/client/http/TestHttpClient.java
./org/projectnessie/client/rest/TestResponseFilter.java
```

Q2: I didn't quite understand this comment - "JDK8 isn't directly supported. Must use separate toolchain" in [637](https://github.com/projectnessie/nessie/pull/637)


Q3: Hive ones were removed from the matrix, was there a reason - [1796](https://github.com/projectnessie/nessie/pull/1796/files#diff-7829468e86c1cc5d5133195b5cb48e1ff6c75e3e9203777f6b2e379d9e4882b3)

Q4: Spark 3.x all use 2.10.0 - IMO, the test matrix seems to be covering the jackson versions (barring jdk version).. do we want more entries here?

cc @snazy

**Comment by rymurr** [comment]
Q1: im referring to the objects and API calls from `HttpTreeApi` adnd `HttpContentsApi` rather than our http client. The goal is to exercise all the objects in `model` and specifically the Jackson annotations there. Does that make more sense?

Q2: I honestly don't remember...should have had more info on the PR.

Q3: Everything Hive related was removed so we didn't think we needed to test them anymore

Q4: I think we need to test every environment Nessie can be used in. This includes:
- spark 3.1.x + delta
- spark 3.x + iceberg
- hive + iceberg
- flink + iceberg
- standalone iceberg
So we should probably add back in the Hive tests and add whatever version flink uses. Iceberg uses (and shades) a newer version of jackson compared to all of these so we should double check that one too

Make sense?
